src/BinSkim.Sdk/CompilerDataLogger.cs

```diff
@@ -260,29 +266,31 @@ public void Write(BinaryAnalyzerContext context, CompilerData compilerData)
                 { "error", string.Empty }
             };
 
-            this.telemetryClient.TrackEvent(CompilerEventName, properties: properties);
-
             if (!string.IsNullOrWhiteSpace(compilerData.CommandLine))
             {
                 string commandLineId = Guid.NewGuid().ToString();
-                properties.Add("commandLineId", commandLineId);
+                properties.Add(CommandLineId, commandLineId);
 
                 SendChunkedContent(CommandLineEventName,
                                    commandLineId,
-                                   "commandLine",
+                                   CommandLine,
                                    compilerData.CommandLine);
             }
 
             if (!string.IsNullOrWhiteSpace(compilerData.AssemblyReferences))
             {
                 string assemblyReferencesId = Guid.NewGuid().ToString();
-                properties.Add("assemblyReferencesId", assemblyReferencesId);
+                properties.Add(AssemblyReferencesId, assemblyReferencesId);
 
                 SendChunkedContent(AssemblyReferencesEventName,
                                    assemblyReferencesId,
-                                   "assemblyReferences",
+                                   AssemblyReferences,
                                    compilerData.AssemblyReferences);
             }
+
+            // To prevent loss of relationship between compiler and assembly/commandline data
```

Comment:
>prevent [](http://example.com/codeflow?start=18&length=7)

This comment would be clearer if it explained how 'loss of relationship' is manifest in the actual data stream. I assume it's because we have no generated ID/guid? can you add just a bit more detail?